solr/core/src/java/org/apache/solr/parser/SolrQueryParserBase.java

```diff
@@ -78,6 +79,14 @@
   static final int MOD_NOT     = 10;
   static final int MOD_REQ     = 11;
 
+  protected ScoreOverlaps scoreOverlaps = ScoreOverlaps.AS_SAME_TERM;
+
+  public static enum ScoreOverlaps {
```

Comment:
the docs on these should be actual javadocs with `{@link classname}` for the implementation classes